utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/universal/UniversalSchemaPayloadDeserializer.scala

```diff
@@ -97,3 +102,55 @@ object PlainTextPayloadDeserializer extends UniversalSchemaPayloadDeserializer {
   }
 
 }
+
+class JsonTypingResultPayloadDeserializer(typingResult: TypingResult) extends UniversalSchemaPayloadDeserializer {
+  import pl.touk.nussknacker.engine.util.json.JsonSchemaImplicits._
+
+  override def deserialize(
+      expectedSchemaData: Option[RuntimeSchemaData[ParsedSchema]],
+      writerSchemaData: RuntimeSchemaData[ParsedSchema],
+      buffer: ByteBuffer
+  ): Any = {
+    val jsonSchema: OpenAPIJsonSchema =
+      expectedSchemaData
+        .getOrElse(writerSchemaData)
+        .schema
+        .asInstanceOf[OpenAPIJsonSchema]
+
+    val bytes = new Array[Byte](buffer.remaining())
+    buffer.get(bytes)
+    val string = new String(bytes, StandardCharsets.UTF_8)
+
+    val inputJson = new JSONTokener(string).nextValue()
+    val validatedJson = jsonSchema
+      .rawSchema()
+      .validateData(inputJson)
+      .valueOr(errorMsg => throw CustomNodeValidationException(errorMsg, None))
```

Comment:
not sure whether it is the right Exception class - it probably was chosen because it is actually used for deserialization errors e.g., in `RequestResponseHttpHandler`